components/builder-core/src/api_client.rs

```diff
@@ -188,8 +186,13 @@ impl ApiClient {
         let dst_file_path = dst_path.join(file_name);
 
         debug!("Writing to {}", &tmp_file_path.display());
-        let mut f = File::create(&tmp_file_path).map_err(Error::IO)?;
-        io::copy(&mut resp, &mut f).map_err(Error::IO)?;
+        let mut f = tokio::fs::File::create(&tmp_file_path).await
+                                                           .map_err(Error::IO)?;
+
+        let mut stream = resp.bytes_stream();
+        while let Some(chunk) = stream.next().await {
+            f.write_all(&chunk?).await.map_err(Error::IO)?;
+        }
```

Comment:
It is crazy there is not an efficient way to write a stream in tokio's `AsyncWriteExt`. Looks like `futures` has an [`into_sink`](https://docs.rs/futures/0.3.5/futures/io/trait.AsyncWriteExt.html#method.into_sink) if we wanted to use that.